lib/chef/guard_interpreter/resource_guard_interpreter.rb

```diff
@@ -106,30 +106,17 @@ def get_interpreter_resource(parent_resource)
         interpreter_resource
       end
 
-      def block_from_attributes(attributes)
-        Proc.new do
-          attributes.each_key do |attribute_name|
-            send(attribute_name, attributes[attribute_name]) if respond_to?(attribute_name)
-          end
-        end
-      end
-
       def merge_inherited_attributes
         inherited_attributes = []
 
         if @parent_resource.class.respond_to?(:guard_inherited_attributes)
           inherited_attributes = @parent_resource.class.send(:guard_inherited_attributes)
         end
 
-        if inherited_attributes && !inherited_attributes.empty?
-          inherited_attributes.each do |attribute|
-            if @parent_resource.respond_to?(attribute) && @resource.respond_to?(attribute)
-              parent_value = @parent_resource.send(attribute)
-              child_value = @resource.send(attribute)
-              if parent_value || child_value
-                @resource.send(attribute, parent_value)
-              end
```

Comment:
This code was potentially hiding a bug, because `nil` and `false` are treated different in resource properties but are considered the same here.